lib/success_response.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'json'
+
+class SuccessResponse
+  attr_reader :error_message
+
+  def initialize(response)
+    @response = response
+    @error_message = nil
+  end
+
+  def ok?
+    body
+    true
+  end
+
+  def body
+    if response.body.empty?
+      @error_message = "I couldn't manage to get any weather info."
+      return {}
+    end
```

Comment:
Why do you insist on this?
Do you want to learn or do want to do your thing?

Reply:
> Why do you insist on this?

'This' is slightly unclear, but since you're referring to lines 19 to 22. this is almost exactly what you only just recommended to me last night.

I'm handling an empty response without triggering an error

> Do you want to learn or do you want to do your thing?

I want to learn whist doing my thing. 😊 If I was just to copy code, I doubt I would learn as much.